### PR TITLE
[CDAP-21093] Add errorCodeType, errorCode & supportedDocURL in mdc tags

### PR DESCRIPTION
**Pull request by itsankit-google**
This PR addresses https://github.com/cdapio/cdap/pull/15764#pullrequestreview-2500763517

### CODE REVIEW

File: cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java [hunk 1]
```diff
@@ -139,6 +140,17 @@ private void addErrorClassificationTags(ILoggingEvent event, Map<String, String>
             ((ProgramFailureException) throwable).getErrorReason());
         modifiableMDC.put(Constants.Logging.TAG_ERROR_TYPE,
             ((ProgramFailureException) throwable).getErrorType().name());
+        String errorCodeType = ((ProgramFailureException) throwable).getErrorCodeType();
+        int errorCode = ((ProgramFailureException) throwable).getErrorCode();
+        String supportedDocURL =
+            ((ProgramFailureException) throwable).getSupportedDocumentationUrl();
+        if (!Strings.isNullOrEmpty(errorCodeType)) {
```
Comment (sahusanket) [inline cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java]: NIT : is it possible that `errorCodeType` is null ? but `errorCode` is not null ? 

in that case should be include it ?

**Review by sahusanket — APPROVED** [review]
Approved with a minor comment

File: cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java (hunk 1 above)
Comment (itsankit-google) [reply cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java]: Nope, it is not possible.

Comment (itsankit-google) [reply cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java]: on a second thought, I can understand error code might not be always `int` so it is better to cater that

Comment (itsankit-google) [reply cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java]: done.

**Pull request merged**